verrazzano/rancher
# VZ-10244 Delete the inactive auth token during refresh

## Problem
Able to access Rancher console even after the session gets timed out in Keycloak and token inactive error messages frequently seen in the Rancher log

## Analysis
Rancher code tries to refresh the token periodically based on `auth-user-info-max-age-seconds` setting. If the Rancher fails to refresh the token, it continuously tries to refresh until it succeeds. If the Session is inactive, it will not be able to refresh the token and the error message is seen frequently in the logs. As the token is not yet deleted, the Rancher console is still accessible.

## Solution
Delete the token while refreshing the token when the session or token is inactive.

## Testing
Manually tested the following scenarios:
1. Delete the Rancher client session from the Keycloak console and observe that the token gets deleted subsequently as we navigate the Rancher console. Re-login is required which creates a new token that provides an access Rancher console.
2. Let the Rancher client session get timed out after 30 minutes by not accessing Rancher console after logging in. Observed that the token gets deleted subsequently and the Rancher console is no longer accessible with the old token. Re-login is required which creates a new token that provides an access Rancher console.
